File: app/src/main/java/code/name/monkey/retromusic/activities/bugreport/model/github/ExtraInfo.java
package code.name.monkey.retromusic.activities.bugreport.model.github;

import java.util.LinkedHashMap;
import java.util.Map;

public class ExtraInfo {

  private final Map<String, String> extraInfo = new LinkedHashMap<>();

  public void put(String key, String value) {
    extraInfo.put(key, value);
  }

  public void put(String key, boolean value) {
    extraInfo.put(key, Boolean.toString(value));
  }

  public void put(String key, double value) {
    extraInfo.put(key, Double.toString(value));
  }

  public void put(String key, float value) {
    extraInfo.put(key, Float.toString(value));
  }

  public void put(String key, long value) {
    extraInfo.put(key, Long.toString(value));
  }

  public void put(String key, int value) {
    extraInfo.put(key, Integer.toString(value));
  }

  public void put(String key, Object value) {
    extraInfo.put(key, String.valueOf(value));
  }

  public void remove(String key) {
    extraInfo.remove(key);
  }

  public String toMarkdown() {
    if (extraInfo.isEmpty()) {
      return "";
    }

    StringBuilder output = new StringBuilder();
    output.append("Extra info:\n" + "---\n" + "<table>\n");
    for (String key : extraInfo.keySet()) {
      output
          .append("<tr><td>")
          .append(key)
          .append("</td><td>")
          .append(extraInfo.get(key))
          .append("</td></tr>\n");
    }
    output.append("</table>\n");

    return output.toString();
  }
}

File: app/src/main/java/code/name/monkey/retromusic/activities/bugreport/model/github/ExtraInfo.kt
package code.name.monkey.retromusic.activities.bugreport.model.github

class ExtraInfo {
    private val extraInfo: MutableMap<String, String> = LinkedHashMap()
    fun put(key: String, value: String) {
        extraInfo[key] = value
    }

    fun put(key: String, value: Boolean) {
        extraInfo[key] = value.toString()
    }

    fun put(key: String, value: Double) {
        extraInfo[key] = value.toString()
    }

    fun put(key: String, value: Float) {
        extraInfo[key] = value.toString()
    }

    fun put(key: String, value: Long) {
        extraInfo[key] = value.toString()
    }

    fun put(key: String, value: Int) {
        extraInfo[key] = value.toString()
    }

    fun put(key: String, value: Any) {
        extraInfo[key] = value.toString()
    }

    fun remove(key: String) {
        extraInfo.remove(key)
    }

    fun toMarkdown(): String {
        if (extraInfo.isEmpty()) {
            return ""
        }
        val output = StringBuilder()
        output.append(
            """
    Extra info:
    ---
    <table>
    
    """.trimIndent()
        )
        for (key in extraInfo.keys) {
            output
                .append("<tr><td>")
                .append(key)
                .append("</td><td>")
                .append(extraInfo[key])
                .append("</td></tr>\n")
        }
        output.append("</table>\n")
        return output.toString()
    }
}